### PR TITLE
Use multipart upload to copy local files to S3

### PR DESCRIPTION
**Pull request by fauvel**
See [this blog post](https://aws.amazon.com/blogs/aws/amazon-s3-multipart-upload/) and [the docs](http://docs.aws.amazon.com/AmazonS3/latest/dev/usingHLmpuPHP.html)

### CODE REVIEW

**Comment by fauvel** [comment]
@njam please review

File: library/CM/File.php [hunk 1]
```diff
@@ -405,20 +405,21 @@ protected function _copyToFileStreaming(CM_File $file, CM_File_Filesystem_Adapte
...
+                'Source adapter'      => get_class($adapterSource),
+                'Destination adapter' => get_class($adapterDestination),
             ]);
         }
-        if (!@fclose($streamDestination)) {
-            throw new CM_Exception('Could not close stream for `' . $file->getPath() . '`');
-        }
-        if (!@fclose($streamSource)) {
-            throw new CM_Exception('Could not close stream for `' . $this->getPath() . '`');
+        if (is_resource($streamSource) && !@fclose($streamSource)) {
```
Comment (fauvel) [inline library/CM/File.php]: Note: The stream is being closed by Guzzle after the transfer completed and the AWS uploader gets destroyed in `CM_File_Filesystem_Adapter_AwsS3::writeStream`, hence this `is_resource` check.

**Comment by njam** [comment]
Makes sense, code looks good.

I'll update with the results from the test upload.

**Comment by njam** [comment]
Works fine!
screen shot 2016-02-26 at 17 50 23

**Pull request merged**